src/praisonai-agents/test_architectural_fixes.py

```
@@ -1,161 +1,161 @@
-    context = await agent._async_build_memory_context("test")
-    assert context == ""  # Should return empty string with no memories
-    print("✓ Async context building works")
+    # Test Agent __del__ method
+    try:
+        agent = Agent(name="test")
+        agent._closed = False  # Simulate agent that hasn't been closed
+        
+        # Test __del__ method exists and works
+        agent.__del__()
+        print("✓ Agent.__del__() performs lightweight cleanup")
+    except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)